desktop-app/src/renderer/components/Previewer/Device/index.tsx

```diff
@@ -199,8 +201,10 @@ const Device = ({ isPrimary, device, setIndividualDevice }: Props) => {
     if (!isIndividualLayout) {
       dispatch(setLayout(PREVIEW_LAYOUTS.INDIVIDUAL));
       setIndividualDevice(selectedDevice);
+      dispatch(setZoom100());
```

Comment:
Instead of `restoreZoom` logic, can you try handling it in the `Device` component to use the `zoomFactor` when in non-individual mode and `individualZoomFactor` when in an individual mode? 

That way, the user set zoom levels for each of those modes will be retained. The user won't be annoyed about the 100% zoom if they don't like it. They can change it once and we'll use them the next time onwards.

Reply:
thanks again Manoj,

I've got it to persist the zoom factor used for the individual view as `individualZoomFactor`. should each device have their own zoomFactor store, or is the individual zoom applicable at a higher level to just the view?

Reply:
One global `individualZoomFactor` should be enough to start with.

Reply:
got it, thanks @manojVivek. I went with your second suggestion, resulting in global `individualZoomFactor` implemented and functioning